enhancements/installer/ibmcloud-ipi.md

```diff
@@ -0,0 +1,181 @@
+  ? Cluster Name test
+  ? Pull Secret [? for help] ****
+  ```
+
+- Implement IBM Cloud specific platform and machine pool installer customizations
+  
+  **Platform**
+  
+  - region (required string): The IBM Cloud region where the cluster will be created.
+  - cisInstanceCRN (required string): The Cloud Internet Services CRN managing the base domain DNS zone.
+  - clusterOSImage (required string): The name of the RHCOS custom image to use for machines.
+  - resourceGroup (optional string): The name of an existing resource group where the cluster and all required resources will be created.
+  - defaultMachinePlatform (optional object): Default machine pool properties that apply to machine pools that do not define their own IBM Cloud specific properties.
+  
+  If one of these is specified, they ALL must be specified.
+  - vpc (optional string): The name of an existing VPC network.
+  - vpcResourceGroup (optional string): The name of the existing VPC's resource group.
+  - subnets (optional array of strings): A list of existing subnet IDs. Leave unset and the installer will create new subnets in the VPC network on your behalf.
+  
+  **Machine Pool**
+  - type (optional string): The VSI machine profile.
+  - zones (optional array of strings): The availability zones used for machines in the pool.
+  - bootVolume (optional object):
+    - encryptionKeyCRN: (optional string): The CRN referencing a a Key Protect or Hyper Protect Crypto Services key to use for volume encryption. If not specified, a provider managed encryption key will be used.
+
+- Provide documentation (in similar format to other supported IPI platforms) to help users use and customize the IPI installer on IBM Cloud:
+  - Prerequisite instructions prior to invoking installer
+  - Description of installer options and customizations as they apply to IBM Cloud
+  - Post installation instructions on further customizations a user may wish to apply
+
+### User Stories
+
+Story 1
+As an OpenShift consumer, I want to quickly, with minimal input and default options, be able to use the OpenShift installer to create and destroy an OpenShift 4 cluster on IBM Cloud.
+
+Story 2
+As an OpenShift consumer, I want to be able to use the OpenShift installer with customizations (i.e. enhanced security) to create and destroy an OpenShift 4 cluster on IBM Cloud.
+
+Story 3
+As an OpenShift platform, I want OpenShift on IBM Cloud to be maintained and released like other supported OpenShift platforms and covered by CI tests.
+
+
+### Implementation Details/Notes/Constraints
+
+- The default IPI-provisioned cluster on IBM Cloud will be a [single region, multizone (3 zones) cluster](https://cloud.ibm.com/docs/containers?topic=containers-ha_clusters#multizone).
+
+- An IBM Cloud RHCOS image must be made available and maintained - ([current 4.7 image location](https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.7/latest/)). The images will be imported as custom images for IBM Cloud.
```

Comment:
I expect the image is probably a similar size as on other platforms no? Maybe worth double checking with the Installer team what we do on other platforms and see what their recommendation is here. IIUC we do something like this on platforms like vSphere as part of the installer so maybe it's not unprecendented?

Reply:
There's a sequence of steps involved in creating a custom VPC image.
1. RHCOS image downloaded, or otherwise accessible from somewhere.
2. The image needs to be uploaded to COS _(COS instance itself and/or bucket created to house the image)_.
3. VPC needs to be granted access to above COS bucket containing rhcos image.
4. VPC custom image then needs to be created based on the rhcos image stored in COS.
The lead time for this process and the amount of other work we have before us lead us to conclude this manual prerequisite was sufficient for an initial IPI MVP.
@staebler thoughts? I'm not familiar with how rhcos images are managed in vSphere.

Reply:
It would be good to have this justification in the body of the enhancement, for readers who come along after the PR merges.

It sounds like the user will have some setup to do before they can use the IPI installer with their IBM Cloud account. Is it just the 4 steps in your comment above, or is there more to it? We don't have a separate user experience section, so maybe you could add a sort description of all of the steps to this section of the doc. Something along the lines of https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md#high-level-end-to-end-workflow (although probably shorter in your case).